1-Draft/RFCXXXX-Send-MailMessage.md

```diff
@@ -0,0 +1,95 @@
+---
+RFC: RFCXXXX
+Author: Travis Plunk
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: cmdlets
+Comments Due: 4/30/2019
+Plan to implement: No
+---
+
+# Obsoleting `Send-MailMessage`
+
+`Send-MailMessage` does not support many modern protocols leading to the inability to use this with modern secure mail services.
+See [DotNet DE0005](https://github.com/dotnet/platform-compat/blob/master/docs/DE0005.md).
+Since the protocols it uses are no longer considered secure, the cmdlet as is should not be used.
+I recommend that we implement a plan to remove the cmdlet from powershell core and, if needed,
+re-implement as a separate module.
+
+## Motivation
+
+### Primary
+
+The underlying API, [`SmtpClient`](https://docs.microsoft.com/dotnet/api/system.net.mail.smtpclient), doesn't support many modern protocols.
+It is compat-only.
+It's great for one off emails from tools, but doesn't scale to modern requirements of the protocol.
+
+### Secondary
+
+    Most major mail platforms now have REST methods to send mail, which allows `Invoke-RestMethod` to allow sending mail messages.
+
+    * gmail
+      * https://developers.google.com/gmail/api/v1/reference/users/messages/send
+      * https://stackoverflow.com/questions/24460422/how-to-send-a-message-successfully-using-the-new-gmail-rest-api
+    * Office365
+      * https://docs.microsoft.com/en-us/previous-versions/office/office-365-api/api/version-2.0/mail-rest-operations#SendMessages
+
+## Obsoletion  Plan
+
+### Add a warning to 6.2
+
+In a 6.2, we have added an obsolete warning to `Send-MailMessage` that an alternative method should be found.
+
+**Note:** Adding an obsolete warning should not break compatibility in any way.
+
+### Positive confirmation of issue in 6.3
+
+We will add a switch requiring the user to accept the risk of using older **possibly** less secure implementations.
+This switch would be called `-AllowUnsecureConnection` and would be mandatory
+
+**Note:** The switch name is based on the [DSC Property](https://docs.microsoft.com/en-us/powershell/dsc/managing-nodes/metaconfig#configuration-server-blocks)
+
+### If needed, develop an alternative
+
+If needed, develop a new cmdlet based on the DotNet recommended solution of [MailKit](https://github.com/jstedfast/MailKit).
```

Comment:
So now we enter the realms of JavaScript with dependencies all the way down.

PowerShell devs and admins will now need to keep track of a PowerShell version, a SMTP module version and MailKit version to ensure they don't have any security issues.

Reply:
Ideally that would fall on the maintainer of the cmdlet/module that's redistributing MailKit (we don't have dependency resolution between Nuget.org and the PS Gallery, so it'd have to be packaged as a binary). That might be us, or it might be someone in the community.